gmso/core/site.py

```diff
@@ -125,14 +124,29 @@ def atom_type(self, val):
         val = _validate_atom_type(val)
         self._atom_type = val
 
+    @property
+    def position(self):
+        return self._position
+
+    @position.setter
+    def position(self, position):
+        self._position = _validate_position(position)
+
     def __repr__(self):
         return "<Site {}, id {}>".format(self.name, id(self))
 
 
 def _validate_position(position):
+    if position is None:
+        return None
```

Comment:
As per the slack discussion we had yesterday, I think the default position should be np.array([np.nan]*3).

Reply:
The site positions are now `u.unyt_array(np.array([np.nan]*3), u.nm)`.